src/example/connect.ts

```diff
@@ -1,37 +1,38 @@
 
 import watchAndNotify from '../index';
-import { MongoDataType, MTROptions, RabbitDataType, DataObjectType } from '../paramTypes';
+import { MongoDataType, MTROptions, RabbitDataType, DataObjectType, middlewareFunc } from '../paramTypes';
 
 // Create two colQCouples and connect to two different queues.
 
 // First colQCouple
+const middleware1: middlewareFunc = (x: DataObjectType) => {
+	const x2: DataObjectType = x;
+	x2.operation += '!!!';
+	return x;
+}
+
 const mongoData1: MongoDataType = {
 	collectionName: 'files',
 	connectionString: 'mongodb://localhost:27017/devDB?replicaSet=rs0',
 };
 const rabbitData1: RabbitDataType = {
-	queueName: 'MyQueueName1',
+	queues: [
+		{name:'MyQueueName1', middleware: middleware1},
+		{name: 'MyQueueName2'}
+	],
 	rabbitURI: 'amqp://localhost'
 };
 
-const options1: Partial<MTROptions> =
-{
-	silent: false,
-	middleware: (x: DataObjectType) => {
-		const x2: DataObjectType = x;
-		x2.operation += '!!!';
-		return x;
-	},
-}
+const options1: Partial<MTROptions> = {silent: false}
 
 // Second colQCouple
 const mongoData2: MongoDataType = {
-	collectionName: 'files',
+	collectionName: 'permissions',
 	connectionString: 'mongodb://localhost:27017/devDB?replicaSet=rs0',
 };
 
 const rabbitData2: RabbitDataType = {
-	queueName: 'MyQueueName2',
+	queues: [{ name:'Queue3'}],
```

Comment:
3?

Reply:
In the previous example I set up two queues, so this is queue 3. Anyway, I changed this back to queue2 in any case this is confusing.